Grievance_Backend/grievance/src/main/java/com/grievance/controller/TicketController.java

```diff
@@ -37,21 +48,63 @@ public class TicketController {
    */
   @PostMapping("/addTicket")
   public ResponseEntity<?> saveTicket(
-     @RequestBody final TicketInDto ticketInDto) {
-     Optional<TicketOutDto> optionalTicketOutDto =
-         ticketService.saveTicket(ticketInDto);
+            @RequestBody final TicketInDto ticketInDto) {
+    Optional<TicketOutDto> optionalTicketOutDto =
+            ticketService.saveTicket(ticketInDto);
     return new ResponseEntity<>(optionalTicketOutDto, HttpStatus.CREATED);
   }
 
   /**
    * controller method to return list of all tickets.
    *
-   * @return Responseentity with optional of list of all tickets.
+   * @param departmentName
+   * @param email
+   * @param password
+   * @return ResponseEntity with optional of list of all tickets.
    */
   @GetMapping("/listAllTickets")
-  public ResponseEntity<?> listAllTickets() {
-      Optional<List<TicketOutDto>>
-      optional = ticketService.listOfAllTickets();
-      return new ResponseEntity<>(optional, HttpStatus.ACCEPTED);
+  public ResponseEntity<?> listAllTickets(
+    @RequestParam(required = false) final String departmentName,
+    @RequestHeader final String email,
+    @RequestHeader final String password
+  ) {
+    try {
+      if (departmentName == null) {
+        authenticatingUser.checkIfUserIsAdmin(email, password);
+        Optional<List<TicketOutDto>> optionalListOfTickets =
+              ticketService.listOfAllTickets();
+        return new ResponseEntity<>(optionalListOfTickets, HttpStatus.ACCEPTED);
+      } else {
+        authenticatingUser.checkIfUserExists(email, password);
+        Optional<List<TicketOutDto>> optionalListOfTickets =
+              ticketService.listOfAllTicketsByDepartmentName(
+          departmentName
+        );
+        return new ResponseEntity<>(optionalListOfTickets, HttpStatus.ACCEPTED);
+      }
+    } catch (UnauthorisedUserException e) {
+      return new ResponseEntity<>("Unauthorised User", HttpStatus.UNAUTHORIZED);
+    }
   }
+
+  /**
+   * controller method to return list of all tickets.
+   * @param ticketId
+   * @param ticketInDto
+   * @return Responseentity with optional of updated TicketOut DTO.
+   */
+  @PutMapping("/update")
+  public ResponseEntity<?> updateTickets(
+          @RequestParam final Integer ticketId,
+          @RequestBody final TicketInDto ticketInDto) {
+      try {
+      Optional<TicketOutDto>  optionalTicketOutDto = ticketService.updateTicket(
+            ticketInDto, ticketId);
+      return new ResponseEntity<>(optionalTicketOutDto, HttpStatus.OK);
```

Comment:
do not return Optional directly, fetch the value from Optional and return that, also change ? to exact Object  in ResponseEntity .

make this changes in all controllers.